seamless-immutable/seamless.d.ts

```diff
@@ -0,0 +1,83 @@
+// modification, are permitted provided that the following conditions are met:
+
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+
+// * Neither the name of seamless-immutable nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+
+declare namespace SeamlessImmutable {
+  interface MergeConfig {
+    deep?: boolean;
+    merger?: Function;
+  }
+
+  interface Options {
+    prototype?: any;
+  }
+
+  interface AsMutableOptions {
+    deep: boolean;
+  }
+
+  export interface ImmutableObject<T> {
+    set(property: string, value: any): ImmutableObject<any>;
+    setIn(propertyPath: Array<string>, value: any): Immutable<any>;
+
+    asMutable(): T;
+    asMutable(opts: AsMutableOptions): T;
+
+    merge(part: any, config?: MergeConfig): ImmutableObject<T>;
+
+    update(property: string, updaterFunction: (value: any) => any): ImmutableObject<T>;
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any) => any, arg1: any): ImmutableObject<T>;
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any) => any, arg1: any, arg2: any): ImmutableObject<T>;
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any) => any, arg1: any, arg2: any, arg3: any): ImmutableObject<T>;
+    update(property: string, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any, additionalParameter4: any) => any, arg1: any, arg2: any, arg3: any, arg4: any): ImmutableObject<T>;
+
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any) => any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any) => any, arg1: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any) => any, arg1: any, arg2: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any) => any, arg1: any, arg2: any, arg3: any): ImmutableObject<T>;
+    updateIn(propertyPath: Array<string>, updaterFunction: (value: any, additionalParameter1: any, additionalParameter2: any, additionalParameter3: any, additionalParameter4: any) => any, arg1: any, arg2: any, arg3: any): ImmutableObject<T>;
+
+    without(property: string): ImmutableObject<any>;
```

Comment:
this is already covered by `without(...properties: string[]): ImmutableObject<any>;`